### PR TITLE
Add Docker role

### PR DESCRIPTION
**Pull request by nicl**
Adds Docker role.

### CODE REVIEW

File: roles/docker/tasks/main.yml [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+- name: Add repo key
+  apt_key:
+    url: "https://download.docker.com/linux/debian/gpg"
+    state: present
+
+- name: Add docker.io repo
+  apt_repository: repo='deb [arch=amd64] https://download.docker.com/linux/debian xenial stable' state=present
+
+- name: Install docker
+  apt: name=docker-ce update_cache=yes state=installed force=yes
```
Comment (sihil) [inline roles/docker/tasks/main.yml]: I really dislike floating versions. Or perhaps: I really dislike debugging problems that end up being caused by an upgrade of something you depend on downstream being implicitly pulled in.

I'd like them more (but still not a whole lot) if we had serverspec to run the finished AMI through its paces before allowing anything else to use it. For now though we should force people to chose a version that we know works.

**Comment by sihil** [comment]
You might want to look at the `packages` role to see if you can do what you want to without a new role. Having said that, docker seems useful enough that a role makes sense.

**Comment by franziskas** [comment]
@nicl @sihil anything to salvage here?

**Comment by AWare** [comment]
Installing docker as a package in #194 doesn't leave the system with a usable docker. 

My instinct is that the docker uninstall in https://github.com/guardian/deploy-tools-platform/blob/master/teamcity/install-docker.sh is required. I'd like to add this to the Docker role and merge it into my PR. Would also parameterise the version while I'm at it. 

@sihil @franziskas @nicl all okay with this?

**Comment by AWare** [comment]
@nicl I've rebased this to test on code.

**Comment by AWare** [comment]
Tested.

**Pull request merged**